dataactbroker/handlers/accountHandler.py

```diff
@@ -137,6 +144,117 @@ def login(self,session):
+                        'maxAttribute:First-Name']
+                    middle_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:Middle-Name']
+                    last_name = max_dict["cas:serviceResponse"]['cas:authenticationSuccess']['cas:attributes'][
+                        'maxAttribute:Last-Name']
+
+                    user.email = email
+
+                    # Check for None first so the condition can short-circuit without
+                    # having to worry about calling strip() on a None object
+                    if middle_name is None or middle_name.strip() == '':
+                        user.name = first_name + " " + last_name
+                    else:
+                        user.name = first_name + " " + middle_name[0] + ". " + last_name
+                    user.user_status_id = UserHandler().getUserStatusId('approved')
+
+                    # If part of the SYS agency, use that as the cgac otherwise use the first agency provided
+                    if list(filter(lambda g: g.endswith("SYS"), cgac_group)):
+                        user.cgac_code = "SYS"
+                        # website admin permissions
+                        user.permissions = 2
+                    else:
+                        user.cgac_code = cgac_group[0][-3:]
+                        # regular user permissions
+                        user.permissions = 1
+
+                    sess.add(user)
+                    sess.commit()
+
+            except MultipleResultsFound:
+                raise ValueError("An error occurred during login.")
+
+            # Create session
+            LoginSession.login(session, user.user_id)
+            permissionList = []
+            for permission in self.interfaces.userDb.getPermissionList():
+                if (self.interfaces.userDb.hasPermission(user, permission.name)):
+                    permissionList.append(permission.permission_type_id)
+            self.interfaces.userDb.updateLastLogin(user)
+            agency_name = self.interfaces.validationDb.getAgencyName(user.cgac_code)
+            return JsonResponse.create(StatusCode.OK, {"message": "Login successful", "user_id": int(user.user_id),
+                                                       "name": user.name, "title": user.title,
+                                                       "agency_name": agency_name,
+                                                       "cgac_code": user.cgac_code, "permissions": permissionList})
+
+        except (TypeError, KeyError, NotImplementedError) as e:
+            # Return a 400 with appropriate message
+            return JsonResponse.error(e,StatusCode.CLIENT_ERROR)
+        except ValueError as e:
+            # Return a 401 for login denied
+            return JsonResponse.error(e,StatusCode.LOGIN_REQUIRED)
```

Comment:
Good question, I will test to ensure it does. The old `login` had it so I was assuming it does, but I will check regardless.